### PR TITLE
Convert the examples to not use the GlobalOpenTelemetry instance.

### PR DESCRIPTION
**Pull request by jkwatson**
I think there is probably still work to be done on cleaning up the story in these examples, since SDK and API usage tends to be mixed together in unfortunate ways.

### CODE REVIEW

File: examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClient.java [hunk 1]
```diff
@@ -116,29 +116,25 @@ public void start(Listener<RespT> responseListener, Metadata headers) {
     }
   }
 
-  private static void initTracing() {
+  private static OpenTelemetry initTracing() {
```
Comment (bogdandrutu) [inline examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClient.java:119]: s/initTracing/initOpenTelemetry?

File: examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClientStream.java [hunk 2]
```diff
@@ -163,6 +162,19 @@ public void start(Listener<RespT> responseListener, Metadata headers) {
     }
   }
 
+  private static OpenTelemetry initTracing(LoggingSpanExporter exporter) {
+    // install the W3C Trace Context propagator
+    // Get the tracer management instance
+    SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder().build();
```
Comment (bogdandrutu) [inline examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClientStream.java:168]: Add a TODO to use the new builder option `addTracerProvider`

File: examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldServer.java [hunk 3]
```diff
@@ -166,25 +166,20 @@ public void onCompleted() {
     }
   }
 
-  private static void initTracing() {
-    // install the W3C Trace Context propagator
-    // Get the tracer management instance
+  private static OpenTelemetry initTracing(LoggingSpanExporter exporter) {
     SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder().build();
     // Set to process the the spans by the LogExporter
     sdkTracerProvider.addSpanProcessor(SimpleSpanProcessor.builder(exporter).build());
```
Comment (bogdandrutu) [inline examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldServer.java:172]: ditto

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java [hunk 4]
```diff
@@ -30,28 +30,26 @@
 public class HttpClient {
 
   // OTel API
+  private static final LoggingSpanExporter loggingExporter = new LoggingSpanExporter();
+  private static final OpenTelemetry openTelemetry = initTracing(loggingExporter);
```
Comment (bogdandrutu) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java:34]: ditto (rename method name)

File: examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java [hunk 5]
```diff
@@ -20,22 +20,72 @@
...
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
 import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.net.InetSocketAddress;
 import java.nio.charset.Charset;
 
 public class HttpServer {
+  // OTel API
+  private static final OpenTelemetry openTelemetry = initTracing(new LoggingSpanExporter());
```
Comment (bogdandrutu) [inline examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java:32]: ditto

File: examples/jaeger/src/main/java/io/opentelemetry/example/jaeger/JaegerExample.java [hunk 6]
```diff
@@ -2,29 +2,26 @@
...
   public JaegerExample(String ip, int port) {
-    this.ip = ip;
-    this.port = port;
+    OpenTelemetrySdk sdk = setupTracing(ip, port);
+    this.sdkTracerManagement = sdk.getTracerManagement();
+    tracer = sdk.getTracer("io.opentelemetry.example.JaegerExample");
   }
 
-  private void setupJaegerExporter() {
+  private OpenTelemetrySdk setupTracing(String ip, int port) {
```
Comment (bogdandrutu) [inline examples/jaeger/src/main/java/io/opentelemetry/example/jaeger/JaegerExample.java:24]: ditto

File: examples/metrics/src/main/java/io/opentelemetry/example/metrics/DoubleCounterExample.java [hunk 7]
```diff
@@ -20,11 +21,11 @@
  * extensions.
  */
 public class DoubleCounterExample {
-
+  private static final OpenTelemetry openTelemetry = DefaultOpenTelemetry.builder().build();
   private static final Tracer tracer =
-      GlobalOpenTelemetry.getTracer("io.opentelemetry.example.metrics", "0.5");
+      openTelemetry.getTracer("io.opentelemetry.example.metrics", "0.31.1");
```
Comment (bogdandrutu) [inline examples/metrics/src/main/java/io/opentelemetry/example/metrics/DoubleCounterExample.java:26]: s/31/13

File: examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClientStream.java (hunk 2 above)
Comment (jkwatson) [reply examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClientStream.java:168]: I assume you mean `addSpanProcessor()`. Yes, we'll need to do this after we publish the next release, but I didn't want to confuse users looking at our docs with TODOs from unreleased changes.

Comment (bogdandrutu) [reply examples/grpc/src/main/java/io/opentelemetry/example/grpc/HelloWorldClientStream.java:168]: K

**Comment by jkwatson** [comment]
I think I got all the methods cleaned up, and the mis-typed arbitrary version number. :)

**Review by bogdandrutu — APPROVED** [review]
FYI we don’t call shutdown in any of the examples, maybe a follow up for that?

**Comment by jkwatson** [comment]
> FYI we don’t call shutdown in any of the examples, maybe a follow up for that?

Yeah, when I have some time, I think I'd like to rethink all of the examples. We have shutdown in a couple of cases (one with the Jaeger exporter, I think?). But, we're not really showing best practices very much in the examples at this point.

**Pull request merged**